scipy/fft/tests/test_czt.py

```diff
@@ -0,0 +1,216 @@
+    lengths = np.random.randint(8, 200, 20)
+    np.append(lengths, 1)
+    for length in lengths:
+        x = np.random.random(length)
+        check_zoomfft(x)
+        check_czt(x)
+
+    # Gauss
+    t = np.linspace(-2, 2, 128)
+    x = np.exp(-t**2/0.01)
+    check_zoomfft(x)
+
+    # Linear
+    x = [1, 2, 3, 4, 5, 6, 7]
+    check_zoomfft(x)
+
+    # Check near powers of two
+    check_zoomfft(range(126-31))
+    check_zoomfft(range(127-31))
+    check_zoomfft(range(128-31))
+    check_zoomfft(range(129-31))
+    check_zoomfft(range(130-31))
+
+    # Check transform on n-D array input
+    x = np.reshape(np.arange(3*2*28), (3, 2, 28))
+    y1 = zoomfft(x, [0, 2-2./28])
+    y2 = zoomfft(x[2, 0, :], [0, 2-2./28])
+    assert_allclose(y1[2, 0], y2, rtol=1e-13, atol=1e-12)
+
+    y1 = zoomfft(x, [0, 2], endpoint=False)
+    y2 = zoomfft(x[2, 0, :], [0, 2], endpoint=False)
+    assert_allclose(y1[2, 0], y2, rtol=1e-13, atol=1e-12)
+
+    # Random (not a test condition)
+    x = np.random.rand(101)
+    check_zoomfft(x)
+
+    # Spikes
+    t = np.linspace(0, 1, 128)
+    x = np.sin(2*np.pi*t*5)+np.sin(2*np.pi*t*13)
+    check_zoomfft(x)
+
+    # Sines
+    x = np.zeros(100, dtype=complex)
+    x[[1, 5, 21]] = 1
+    check_zoomfft(x)
+
+    # Sines plus complex component
+    x += 1j*np.linspace(0, 0.5, x.shape[0])
+    check_zoomfft(x)
+
+
+def test_large_prime_lengths():
+    np.random.seed(0)  # Deterministic randomness
+    for N in (101, 1009, 10007):
+        x = np.random.rand(N)
+        y = fft(x)
+        y1 = czt(x)
+        assert_allclose(y, y1, rtol=1e-12)
+
+
+@dec.slow
+def test_czt_vs_fft():
+    np.random.seed(123)
+    random_lengths = np.random.exponential(100000, size=10).astype('int')
+    for n in random_lengths:
+        a = np.random.randn(n)
+        assert_allclose(czt(a), fft(a), rtol=1e-11)
+
+
+def test_empty_input():
+    assert_raises(ValueError, czt, [])
+    assert_raises(ValueError, zoomfft, [], 0.5)
+
+
+def test_0_rank_input():
+    assert_raises(IndexError, czt, 5)
+    assert_raises(IndexError, zoomfft, 5, 0.5)
+
+
+def test_czt_math():
+    for impulse in ([0, 0, 1],
+                    [0, 0, 1, 0, 0],
+                    np.concatenate(([0, 0, 1], np.zeros(100)))):
+        for m in (1, 3, 5, 8, 101, 1021):
+            for a in (1, 2, 0.5, 1.1):
+                for w in (None, 0.7+0.7j):
+                    # z-transform of an impulse is 1 everywhere
+                    assert_allclose(czt(impulse[2:], m=m, a=a),
+                                    np.ones(m), rtol=1e-14)
+
+                    # z-transform of a delayed impulse is z**-1
+                    assert_allclose(czt(impulse[1:], m=m, a=a),
+                                    czt_points(m=m, a=a)**-1, rtol=1e-14)
+
+                    # z-transform of a 2-delayed impulse is z**-2
+                    assert_allclose(czt(impulse, m=m, a=a),
+                                    czt_points(m=m, a=a)**-2, rtol=1e-14)
```

Comment:
I can't see the `w` parameter tested anywhere here even though it's being looped over.